backend/services/team.py

```diff
@@ -90,7 +95,22 @@ def teams_to_df(self, teams: list[Team]) -> pl.DataFrame:
             team_dfs.append(self.team_to_df(team))
         return pl.concat(team_dfs)
 
-    def update_team(self, team: Team) -> Team:
+    def team_to_team_data(self, team: Team) -> TeamData:
```

Comment:
Question: Did you run into errors when trying do have pydantic handle this? Like as far as I know it *should* just be able to do the transformation for you.